pkg/auth/cred.go

```diff
@@ -17,19 +17,42 @@ limitations under the License.
 package auth
 
 import (
+	"context"
 	"fmt"
+	"time"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"k8s.io/klog/v2"
+	"knative.dev/pkg/logging"
 )
 
+type TokenWrapper struct {
```

Comment:
nit: can we document this token wrapper usage?

Reply:
Also i would suggest rename it to a meaningful name to reflect the behavior.

Reply:
and to avoid exposing this type to the public caller, i would suggest to declare this as a private type

Reply:
done